src/raftstore/store/worker/apply.rs

```diff
@@ -1932,16 +2087,63 @@ pub struct ApplyBatch {
     start: Instant,
 }
 
+pub struct AppliesRes {
+    delegates: Vec<ApplyDelegate>,
+    merged_regions: Vec<u64>,
+    apply_res: Vec<ApplyRes>,
+}
+
+pub struct Suicide {
+    pub region_id: u64,
+    pub term: u64,
+}
+
+pub struct RegionDestroy {
+    pub region_id: u64,
+}
+
+pub struct BorrowDelegate {
+    pub target: u64,
+    pub tx: oneshot::Sender<ApplyDelegate>,
+}
+
+pub enum RegionTask {
+    Apply((Apply, Instant)),
+    Destroy(RegionDestroy),
+    Suicide(Suicide),
+    Proposal(RegionProposal),
+    Borrow(BorrowDelegate),
+    Stop((u64, Sender<()>)),
+}
+
+impl RegionTask {
+    pub fn apply(apply: Apply, start: Instant) -> RegionTask {
+        RegionTask::Apply((apply, start))
+    }
+
+    pub fn proposal(proposal: RegionProposal) -> RegionTask {
+        RegionTask::Proposal(proposal)
+    }
+
+    pub fn destroy(region_id: u64) -> RegionTask {
+        RegionTask::Destroy(RegionDestroy { region_id })
+    }
+
+    pub fn suicide(region_id: u64, term: u64) -> RegionTask {
+        RegionTask::Suicide(Suicide { region_id, term })
+    }
+}
+
 pub struct Destroy {
-    region_id: u64,
+    pub region_id: u64,
 }
 
-/// region related task.
 pub enum Task {
     Applies(ApplyBatch),
     Registration(Registration),
     Proposals(Vec<RegionProposal>),
     Destroy(Destroy),
+    Borrow(BorrowDelegate),
```

Comment:
When will we have borrow task and what's the meaning of "borrow"?

Reply:
When too regions need to merge.

Reply:
Please add a documentation comment on this one. (Hopefully we can add it for the others later.)